src/lib/Scenes/Artwork/Components/CommercialButtons.tsx

```diff
@@ -0,0 +1,226 @@
+                      type
+                      code
+                      data
+                    }
+                  }
+                }
+              }
+            }
+          `,
+          variables: {
+            input: {
+              artworkId: internalID,
+              editionSetId: editionSetID,
+            },
+          },
+          onCompleted: data => {
+            this.setState({ isCommittingCreateOrderMutation: false }, () => {
+              const {
+                commerceCreateOrderWithArtwork: { orderOrError },
+              } = data
+              if (orderOrError.error) {
+                this.onMutationError(orderOrError.error)
+              } else {
+                SwitchBoard.presentModalViewController(this, `/orders/${orderOrError.order.internalID}`)
+              }
+            })
+          },
+          onError: this.onMutationError,
+        })
+      }
+    })
+  }
+
+  handleCreateOfferOrder() {
+    const { relay, artwork, editionSetID } = this.props
+    const { isCommittingCreateOfferOrderMutation } = this.state
+    const { internalID } = artwork
+    if (isCommittingCreateOfferOrderMutation) {
+      return
+    }
+    this.setState({ isCommittingCreateOfferOrderMutation: true }, () => {
+      if (relay && relay.environment) {
+        commitMutation<CommercialButtonsOfferOrderMutation>(relay.environment, {
+          mutation: graphql`
+            mutation CommercialButtonsOfferOrderMutation($input: CreateOfferOrderWithArtworkInput!) {
+              ecommerceCreateOfferOrderWithArtwork(input: $input) {
+                orderOrError {
+                  ... on OrderWithMutationSuccess {
+                    __typename
+                    order {
+                      internalID
+                      mode
+                    }
+                  }
+                  ... on OrderWithMutationFailure {
+                    error {
+                      type
+                      code
+                      data
+                    }
+                  }
+                }
+              }
+            }
+          `,
+          variables: {
+            input: {
+              artworkId: internalID,
+              editionSetId: editionSetID,
+            },
+          },
+          onCompleted: data => {
+            this.setState({ isCommittingCreateOfferOrderMutation: false }, () => {
+              const {
+                ecommerceCreateOfferOrderWithArtwork: { orderOrError },
+              } = data
+              if (orderOrError.error) {
+                this.onMutationError(orderOrError.error)
+              } else {
+                SwitchBoard.presentModalViewController(this, `/orders/${orderOrError.order.internalID}`)
+              }
+            })
+          },
+          onError: this.onMutationError,
+        })
+      }
+    })
+  }
+
+  handleInquiry = () => {
+    SwitchBoard.presentModalViewController(this, `/inquiry/${this.props.artwork.slug}`)
```

Comment:
Is there a good reason to use `slug` here instead of `internalID`?

Reply:
Also, I don't recall the internals but should we also handle an edition-set picker for inquiries and pass along the edition set ID accordingly?

Reply:
I don't know the answer to the second question, what or who would be a good reference for that?

Reply:
I think of `slug` as an external ID which should be used for URL's and similar client facing situations. Also this is the URL used elsewhere, as the routing works.

Reply:
Since this is all internal routing within the app, it seems that `internalID` would be more appropriate? Not a big deal in any case. For other places where we might benefit from caching or other niceties, we may prefer `internalID` out of necessity but I doubt that would come into play much for the inquiry flow.